docs/source/workflow/generate-types.mdx

```diff
@@ -0,0 +1,205 @@
+
+Import the `Resolvers` type into the file where you define your resolvers:
+
+```ts title="resolvers.ts"
+// This is the file where our generated types live
+// (specified in our `codegen.yml` file)
+import { Resolvers } from './__generated__/resolvers-types';
+```
+
+You can now add the `Resolvers` type directly to your resolver map:
+
+```ts
+export const resolvers: Resolvers = {}
+```
+
+Your resolvers can now type check that the *arguments* and *return value* for each resolver match the schema:
+
+```ts
+export const resolvers: Resolvers = {
+  Query: {
+    // TypeScript now complains about the below resolver because
+    // the data returned by this resolver doesn't match the schema type
+    // (i.e., type Query { books: [Book] })
+    books: () => {
+      return "apple";
+    },
+  },
+}
+```
+
+### Context typing for resolvers
+
+You can also configure GraphQL Code Generator to add a type for the context your resolvers share, ensuring TypeScript warns you if you attempt to use a context value that doesn't exist.
+
+To do this, you must first export the interface you pass to Apollo Server as a generic type parameter for typing your context value:
+
+```ts title="src/index.ts"
+// highlight-start
+export interface MyContext {
+  dataSources: {
+    books: Book[];
+  };
+}
+// highlight-end
+
+const server = new ApolloServer<MyContext>({
+  typeDefs,
+  resolvers,
+});
+```
+
+Remember the `contextType` from our [`codegen.yml` file above](#installing-and-configuring-dependencies)? You can pass your exported context interface to the `contextType` configuration option, like so:
+
+```yaml
+# ...
+config:
+  useIndexSignature: true
+  # Providing our context's interface ensures our
+  # context's type is set for all of our resolvers. 
+
+  # Note, this file path starts from the location of the 
+  # file where you generate types. 
+  # (i.e., `/src/__generated__/resolvers-types.ts` above) 
+  contextType: ../index#MyContext # highlight-line
+```
+
+Once you regenerate your types, your context is now automatically typed in all of your resolvers:
+
+```ts
+const resolvers: Resolvers = {
+  Query: {
+    // Our third argument (`contextValue`) has a type here, so we
+    // can check the properties within our resolver's shared context value.
+    books: (_, __, contextValue) => {
+      return contextValue.dataSources.books;
+    },
+  },
+}
+```
+
+## Basic runnable example
+
+Check out our example using Apollo Server with generated types on CodeSandbox:
+
+<a href="https://codesandbox.io/s/github/apollographql/docs-examples/tree/main/apollo-server/v4/generated-types?fontsize=14&hidenavigation=1&initialpath=%2Fgraphql&theme=dark">
+  <img
+    alt="Edit server-generated-types-as4"
+    src="https://codesandbox.io/static/img/play-codesandbox.svg"
+  />
+</a>
+<br/>
+
```

Comment:
The following further reading could be useful to most users:

- How to get types to work with Prisma: https://the-guild.dev/blog/graphql-code-generator-and-prisma
- How to properly configure Models types: https://the-guild.dev/blog/better-type-safety-for-resolvers-with-graphql-codegen#use-your-model-types

Reply:
These seem a bit arbitrary and specific. It looks like you have a whole list of [further reading](https://www.the-guild.dev/graphql/codegen/docs/guides/further-reading) on the docs site. So for a "further reading" section I would rather either link them there, or just directly to the docs. What do you think @rkoron007?

Reply:
We do have a few direct links to the GraphQL Code Generator docs throughout this article. I like your suggestion of plugging the "further reading" section at the end of this article, I'll open a PR for that

Reply:
@rkoron007 Thank you for considering our feedback 😊

@trevor-scheer I've listed relevant links to your documentation page (typescript resolvers generator).
The "further reading" page of codegen documentation links to pages covering a larger scope of code generation features (front-end flavors, high-level articles, etc.)